services/secrets/secrets.go

```diff
@@ -6,43 +6,68 @@ package secrets
 import (
 	"context"
 
+	audit_model "code.gitea.io/gitea/models/audit"
 	"code.gitea.io/gitea/models/db"
+	repo_model "code.gitea.io/gitea/models/repo"
 	secret_model "code.gitea.io/gitea/models/secret"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/services/audit"
 )
 
-func CreateOrUpdateSecret(ctx context.Context, ownerID, repoID int64, name, data string) (*secret_model.Secret, bool, error) {
+func CreateOrUpdateSecret(ctx context.Context, doer, owner *user_model.User, repo *repo_model.Repository, name, data string) (*secret_model.Secret, bool, error) {
 	if err := ValidateName(name); err != nil {
 		return nil, false, err
 	}
 
-	s, err := db.Find[secret_model.Secret](ctx, secret_model.FindSecretsOptions{
-		OwnerID: ownerID,
-		RepoID:  repoID,
+	ss, err := db.Find[secret_model.Secret](ctx, secret_model.FindSecretsOptions{
+		OwnerID: tryGetOwnerID(owner),
+		RepoID:  tryGetRepositoryID(repo),
 		Name:    name,
 	})
 	if err != nil {
 		return nil, false, err
 	}
 
-	if len(s) == 0 {
-		s, err := secret_model.InsertEncryptedSecret(ctx, ownerID, repoID, name, data)
+	if len(ss) == 0 {
+		s, err := secret_model.InsertEncryptedSecret(ctx, tryGetOwnerID(owner), tryGetRepositoryID(repo), name, data)
 		if err != nil {
 			return nil, false, err
 		}
+
+		audit.Record(ctx,
+			auditActionSwitch(owner, repo, audit_model.UserSecretAdd, audit_model.OrganizationSecretAdd, audit_model.RepositorySecretAdd),
+			doer,
+			auditScopeSwitch(owner, repo),
+			s,
+			"Added secret %s.",
```

Comment:
I recommend two extra methods where one outputs `"user"`,`"organization"`, `"repo"`, and the other outputs the pretty printed name (username for users/orgs, repo.FullName for repos) that also take `owner, repo` as params, and then adds this info to the audit log.
So, overall, this line should then look like
```suggestion
			"Added secret %s for %s %s.",
```